gdal/src/main/scala/geotrellis/contrib/vlm/gdal/GDALDataset.scala

```diff
@@ -42,7 +42,7 @@ case class GDALDataset(token: Long) extends AnyVal {
     val width_height = Array.ofDim[Int](2)
     if (GDALWarp.get_transform(token, dataset, numberOfAttempts, transform) <= 0 ||
       GDALWarp.get_width_height(token, dataset, numberOfAttempts, width_height) <= 0)
```

Comment:
What are the possible values and semantics of the value returned integer value? If it can return values other than zero and one, I'd propose having the code included in the exception message so we can at least go look it up in the GDAL docs. We were getting errors thrown here that were due to S3 authentication errors, but at first it looked like there was a data error. Error codes could help ameliorate this.

Reply:
Yeh, according to [this](https://github.com/geotrellis/gdal-warp-bindings/blob/58d496521695b4bae2b7f7ab7d1f17140be0a569/src/bindings.cpp#L511-L512) the return value has some sort of meaning one could inspect for debugging.

Reply:
@metasim I think it returns a number of the datasets that `GDALWarp.method` touched to return a value requested from by the `method`. As @jamesmcclain mentioned right now, that the idea was that probably it would be possible to use this information in some dynamic performance tuning. There is an issue https://github.com/geotrellis/gdal-warp-bindings/issues/52 about it.

Reply:
wooops @metasim mb you're right; @jbouffard can you check that it returns GDAL error codes? As it can return smth internal?

Reply:
@pomadchin Do you have any malformed GeoTiffs? I don't think I have anything that would fail to be read

Reply:
@jbouffard you can point to a non existent file or to a file that is not a geotiff though have a `.tiff` extension

Reply:
@pomadchin @metasim This is what's returned when trying to read a bad file with GDAL:

```
scala> GDALWarp.get_band_count(token, GDALWarp.WARPED, numberOfAttempts, count)
ERROR 4: ../../blah.tif: No such file or directory
res5: Int = -1
```

So while he `ERROR` message shown is correct, the return of `-1` is not.

Reply:
@jbouffard it looks like the only thing we can do in this case is to return to this problem, once it would be addressed in bindings. The best thing we can do now is to write our messages without codes, since it is impossible to retrieve that information from the native interface now.